xds/src/main/java/io/grpc/xds/XdsEndpointResource.java

```diff
@@ -270,4 +293,29 @@ public String toString() {
               .toString();
     }
   }
+
+  static class AddressMetadataParser implements MetadataValueParser {
+
+    @Override
+    public String getTypeUrl() {
+      return "type.googleapis.com/envoy.config.core.v3.Address";
+    }
+
+    @Override
+    public String parse(Any any) throws InvalidProtocolBufferException {
+      Address address = any.unpack(Address.class);
+      SocketAddress socketAddress = address.getSocketAddress();
+      if (socketAddress.getAddress().isEmpty()) {
+        throw new InvalidProtocolBufferException("Address field is empty or invalid.");
```

Comment:
There's no context for this error to know what message was messed up. The metadata parsing code doesn't add context (e.g., the key name) and this isn't mentioning "envoy.config.core.v3.Address" or related messages.